src/ServicePulse.Host/vue/src/components/failedmessages/AllDeletedMessages.vue

```diff
@@ -12,22 +12,26 @@ import MessageList from "./MessageList.vue";
 import ConfirmDialog from "../ConfirmDialog.vue";
 import PaginationStrip from "../../components/PaginationStrip.vue";
 import moment from "moment";
+import type { Ref } from "vue";
+import { FailedMessageViewWithExtendedUIProperties } from "@/resources/FailedMessageView";
+import Configuration from "@/resources/Configuration";
+import { TYPE } from "vue-toastification";
 
 let pollingFaster = false;
-let refreshInterval = undefined;
+let refreshInterval: ReturnType<typeof setInterval>;
 const perPage = 50;
-const configuration = ref([]);
+const configuration: Ref<Configuration | null> = ref(null);
 
 const route = useRoute();
-const groupId = ref(route.params.groupId);
+const groupId: Ref<string | string[]> = ref(route.params.groupId);
 const groupName = ref("");
 const pageNumber = ref(1);
 const totalCount = ref(0);
 const cookies = useCookies().cookies;
 const selectedPeriod = ref("Deleted in the last 7 days");
```

Comment:
The value options for this ref, should be an enum or a specific set of strings, eg `type Period = "All Deleted" | "Deleted in the last 2 Hours" | "Deleted in the last 1 Day" | "Deleted in the last 7 days";`

Reply:
It looks like you already have an array with allowed values, so maybe you could use that.

Reply:
not the groupId ref, but the selectedPeriod ref